protocol/packages/swap/Cargo.toml

```diff
@@ -16,50 +16,77 @@ combinations = [
 feature-groups = { testing = { members = ["testing"], at-least-one = false, mutually-exclusive = false } }
 
 [features]
-# TODO get rid of main and test and specialize astroport into ~_main and ~_test when the CI builds protocols provided as input
-# instead of implicitly building them in matrix
-main = []
-test = []
-
-astroport = [
+dex-astroport_test = [
+    "dep:astroport",
+    "dep:currency",
+    "dep:dex",
+    "dep:finance",
+    "dep:oracle",
+    "dep:platform",
+    "dep:sdk",
     "sdk/cosmos_ibc",
+    "dep:serde",
+]
+dex-astroport_main = [
     "dep:astroport",
     "dep:currency",
     "dep:dex",
     "dep:finance",
     "dep:oracle",
     "dep:platform",
     "dep:sdk",
+    "sdk/cosmos_ibc",
     "dep:serde",
 ]
-osmosis = [
-    "sdk/cosmos",
-    "dep:osmosis-std",
+dex-osmosis = [
     "dep:currency",
     "dep:dex",
     "dep:finance",
     "dep:oracle",
+    "dep:osmosis-std",
     "dep:platform",
     "dep:sdk",
+    "sdk/cosmos",
     "dep:serde",
 ]
 
-testing = ["finance/testing", "dep:serde", "dex/testing"]
+testing = [
+    "dep:currency",
+    "currency/testing",
+    "dep:dex",
+    "dex/testing",
+    "dep:finance",
+    "finance/testing",
+    "dep:oracle",
+    "oracle/stub_swap",
+    "dep:platform",
+    "dep:sdk",
+    "sdk/cosmos_ibc",
+    "dep:prost",
+    "dep:serde",
+]
 
 [dependencies]
-currency = { workspace = true, optional = true }
 dex = { workspace = true, optional = true }
+oracle = { workspace = true, optional = true }
+
+currency = { workspace = true, optional = true }
 finance = { workspace = true, optional = true }
-oracle = { workspace = true, optional = true, features = ["stub_swap"] }
```

Comment:
are you sure "stub_swap" is not needed in all the features? I am

Reply:
You were right, it is needed but only for the `testing` feature, thus I have moved it here:
https://github.com/nolus-protocol/nolus-money-market/pull/422/files/f338d2c5e7996e9a1d7d5d94ff1cb4ddc403f480#diff-eb2041f1f3c24c33db9764eb04a011b1818b14e34b3906f0fd713b4ae491124fR61

Reply:
impossible, the imports [here ](https://github.com/nolus-protocol/nolus-money-market/blob/7522439e31d512f63ef5b05d144ae205938ea324/protocol/packages/swap/src/astroport/mod.rs#L12) are included only under "stub_swap"

Reply:
Then some other package must be including the feature. What I'm trying to say is that it still works without it. Saying "impossible" doesn't exactly contribute much when I know for a fact that it still works.

I will add it to the `dex-astroport_{main|test}` features, but it should probably be investigated who still pulls it.